### PR TITLE
Add Python 3.10 to the pylint matrix

### PR DESCRIPTION
**Pull request by narrieta**
I also added many suppresions of deprecated-method/deprecated-module. Those need to be addressed, but I will do it on separate PRs to facilitate the reviews.

I created 3.10.pylintrc out of 3.6.pylintrc with a few adjustments for new warnings that were added or old warnings that were removed on 3.10. I'll highlight those with comments in the review. Documentation for warnings (use search for specific warning): https://pylint.pycqa.org/en/latest/tutorial.html

### CODE REVIEW

File: .github/workflows/ci_pr.yml [hunk 1]
```diff
@@ -14,8 +14,8 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - python-version: 2.6
-          - python-version: 3.4
+          - python-version: "2.6"
```
Comment (narrieta) [inline .github/workflows/ci_pr.yml:17]: we were using numbers for the version, so "3.10" was actually 3.1

File: ci/3.10.pylintrc [hunk 2]
```diff
@@ -0,0 +1,40 @@
+[MESSAGES CONTROL]
+
+disable=C, # (C) convention, for programming standard violation
+    broad-except, # W0703: *Catching too general exception %s*
+    broad-exception-raised, # W0719: Raising too general exception: Exception
```
Comment (narrieta) [inline ci/3.10.pylintrc:5]: new

File: ci/3.10.pylintrc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    consider-using-dict-comprehension, # R1717: *Consider using a dictionary comprehension*
+    consider-using-from-import, # R0402: Use 'from foo import bar' instead
```
Comment (narrieta) [inline ci/3.10.pylintrc:7]: new

File: ci/3.10.pylintrc [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    consider-using-in, # R1714: *Consider merging these comparisons with "in" to %r*
+    consider-using-set-comprehension, # R1718: *Consider using a set comprehension*
+    consider-using-with, # R1732: *Emitted if a resource-allocating assignment or call may be replaced by a 'with' block*
+    duplicate-code, # R0801: *Similar lines in %s files*
+    fixme, # Used when a warning note as FIXME or TODO is detected
+    logging-format-interpolation, # W1202: Use lazy % formatting in logging functions
+    logging-fstring-interpolation, # W1203: Use lazy % or .format() formatting in logging functions
```
Comment (narrieta) [inline ci/3.10.pylintrc:14]: 13 & 14 new

File: azurelinuxagent/ga/cgroupconfigurator.py [hunk 5]
```diff
@@ -184,10 +184,6 @@ def initialize(self):
 
                 _log_cgroup_info("systemd version: {0}", systemd.get_version())
 
-                # This is temporarily disabled while we analyze telemetry. Likely it will be removed.
```
Comment (narrieta) [inline azurelinuxagent/ga/cgroupconfigurator.py:187]: we added those when we started using the 'azure' slice and drop in files... that has been stable for a long time, so no longer needed

File: .github/workflows/ci_pr.yml [hunk 6]
```diff
@@ -133,13 +136,13 @@ jobs:
         pylint $PYLINTOPTS --jobs=0 $PYLINTFILES
 
     - name: Test with nosetests
-      if: success() || (failure() && steps.install-dependencies.outcome == 'success')
+      if: matrix.python-version != '3.10' && (success() || (failure() && steps.install-dependencies.outcome == 'success'))
```
Comment (narrieta) [inline .github/workflows/ci_pr.yml:139]: we are note testing on 3.10 yet

File: .github/workflows/ci_pr.yml [hunk 7]
```diff
@@ -86,22 +86,25 @@ jobs:
...
+          - python-version: "3.8"
             PYLINTOPTS: "--rcfile=ci/3.6.pylintrc --ignore=tests_e2e"
 
-          - python-version: 3.9
+          - python-version: "3.9"
             PYLINTOPTS: "--rcfile=ci/3.6.pylintrc"
             additional-nose-opts: "--with-coverage --cover-erase --cover-inclusive --cover-branches --cover-package=azurelinuxagent"
 
+          - python-version: "3.10"
+            PYLINTOPTS: "--rcfile=ci/3.10.pylintrc --ignore=tests"
```
Comment (narrieta) [inline .github/workflows/ci_pr.yml:106]: the unit tests are not Python 3.10 compatible, no need to lint them

**Review by maddieford — APPROVED** [review]

**Review by nagworld9 — APPROVED** [review]

File: ci/3.10.pylintrc (hunk 3 above)
Comment (nagworld9) [reply ci/3.10.pylintrc:7]: where do we get this data?

Comment (narrieta) [reply ci/3.10.pylintrc:7]: what do you mean?

Comment (nagworld9) [reply ci/3.10.pylintrc:7]: I mean how do one get these pylint warnings list. Tomorow If I add for 3.11, where do I go and get these?

Comment (narrieta) [reply ci/3.10.pylintrc:7]: I see. 

Run pylint on the code, check the documentation for what each warning means, then make a judgement call: fix the ones that *must* be fixed, for the rest fix as many as possible. When there are too many to fix, think how we address them. Do we want to suppress each one of them or suppress globally? If globally, add to this file.

In this specific case, this warning is just complaining about the syntax used to do the import. We have way too many of these and the functionality/correctness of the code is not affected. It'd be nice to fix them, but I didn't consider spending time on them was worth it so I suppressed globally.

**Pull request merged**